services/server/src/server/services/compiler/lambda/SolcLambda.ts

```diff
@@ -44,45 +44,56 @@ export class SolcLambda implements ISolidityCompiler {
     logger.debug("Compiling with Lambda", { version });
     const response = await this.invokeLambdaFunction(param);
     logger.debug("Compiled with Lambda", { version });
-    const responseObj = this.parseCompilerOutput(response);
-    logger.silly("Lambda function response", { responseObj });
-    return responseObj;
+    logger.silly("Lambda function response", { response });
+    return response;
   }
 
-  private async invokeLambdaFunction(payload: string): Promise<any> {
-    const params: InvokeCommandInput = {
+  private async invokeLambdaFunction(payload: string): Promise<CompilerOutput> {
+    const params: InvokeWithResponseStreamCommandInput = {
       FunctionName: config.get("lambdaCompiler.functionName") || "compile",
       Payload: payload,
     };
 
-    const command = new InvokeCommand(params);
+    const command = new InvokeWithResponseStreamCommand(params);
     const response = await this.lambdaClient.send(command);
 
-    if (!response.Payload) {
+    if (!response.EventStream) {
       throw new Error(
-        "Error: No response payload received from Lambda function"
+        "Error: No response stream received from Lambda function"
       );
     }
 
-    if (response.FunctionError) {
-      const errorObj: { errorMessage: string; errorType: string } = JSON.parse(
-        Buffer.from(response.Payload).toString("utf8")
-      );
-      logger.error("Error invoking Lambda function", {
-        errorObj,
+    let streamResult = "";
+    for await (const event of response.EventStream) {
+      if (event.InvokeComplete?.ErrorCode) {
+        logger.error("Error invoking Lambda function", {
+          errorCode: event.InvokeComplete.ErrorCode,
+          errorDetails: event.InvokeComplete.ErrorDetails,
+          logResult: event.InvokeComplete.LogResult,
+          lambdaRequestId: response.$metadata.requestId,
+        });
+        throw new Error(
+          `AWS Lambda error: ${event.InvokeComplete.ErrorCode} - ${event.InvokeComplete.ErrorDetails} - lamdbaRequestId: ${response.$metadata.requestId}`
+        );
+      } else if (event.PayloadChunk?.Payload) {
+        streamResult += Buffer.from(event.PayloadChunk.Payload).toString(
+          "utf8"
+        );
+      }
+    }
+    logger.silly("Received stream response", { streamResult });
+
+    const output = JSON.parse(streamResult);
```

Comment:
Maybe it could make sense to try catch this and in the case of an error `logger.error` the `streamResult`?